src/github-handler/comment-handler/get-hunk-scope-handler/remote-patch-ranges-handler.ts

```diff
@@ -78,62 +78,50 @@ index cac8fbc..87f387c 100644
 `;
 
 /**
- * Given the patch text (for a whole file) for each file,
- * get each file's hunk's (part of a file's) range
- * @param {Map<string, string>} validPatches patch text from the remote github file
- * @returns {Map<string, Range[]>} the range of the remote patch
- */
-export function patchTextToRanges(
-  validPatches: Map<string, string>
-): Map<string, Range[]> {
-  const allValidLineRanges: Map<string, Range[]> = new Map<string, Range[]>();
-  validPatches.forEach((patch, filename) => {
-    // get each hunk range in the patch string
-    try {
-      const validHunks = parseHunks(_DIFF_HEADER + patch);
-      // TODO: consider returning hunks instead
-      const validLineRanges = validHunks.map(hunk => {
-        return {start: hunk.newStart, end: hunk.newEnd};
-      });
-      allValidLineRanges.set(filename, validLineRanges);
-    } catch (err) {
-      logger.info(
-        `Failed to parse the patch of file ${filename}. Resuming parsing patches...`
-      );
-    }
-  });
-  return allValidLineRanges;
-}
-
-/**
- * For a pull request, get each remote file's current patch range to identify the scope of each patch as a Map,
- * as well as a list of files that cannot have suggestions applied to it within the Pull Request.
- * The list of files are a subset of the total out-of-scope files.
+ * For a pull request, get each remote file's current patch range to identify the scope of each patch as a Map.
  * @param {Octokit} octokit the authenticated octokit instance
  * @param {RepoDomain} remote the remote repository domain information
  * @param {number} pullNumber the pull request number
  * @param {number} pageSize the number of files to return per pull request list files query
- * @returns {Promise<Object<Map<string, Range[]>, string[]>>} the scope of each file in the pull request and the list of files whose patch data could not be resolved
+ * @returns {Promise<Map<string, Hunk[]>>} the scope of each file in the pull request
  */
-export async function getPullRequestScope(
+export async function getPullRequestHunks(
   octokit: Octokit,
   remote: RepoDomain,
   pullNumber: number,
   pageSize: number
-): Promise<{validFileLines: Map<string, Range[]>; invalidFiles: string[]}> {
-  try {
-    const {patches, filesMissingPatch} = await getCurrentPullRequestPatches(
-      octokit,
-      remote,
-      pullNumber,
-      pageSize
-    );
-    const validFileLines = patchTextToRanges(patches);
-    return {validFileLines, invalidFiles: filesMissingPatch};
-  } catch (err) {
+): Promise<Map<string, Hunk[]>> {
+  const files = (
```

Comment:
If you use octokit's iterator API, you can do something along the lines of:

```
for await () {}
```

And avoid pagination concerns if there are more than 100 files.